nixos/modules/services/network-filesystems/webdav.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.webdav;
+in
+{
+  options = {
+    services.webdav = {
+      enable = mkEnableOption "WebDAV server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webdav;
+        description = "The WebDAV package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "User account under which WebDAV runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "Group under which WebDAV runs.";
+      };
+
+      settings = mkOption {
+        type = types.anything;
+        default = { };
+        description = "Attrset that is converted and passed as config file.";
```

Comment:
Please link to upstream documentation. Additionally, please make special note that the configuration file supports storing credentials in an external file, which can be provided by [EnvironmentFile](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=).